src/Http/Headers/src/HttpHeaderParser.cs

```diff
@@ -84,7 +84,7 @@ protected virtual bool TryParseValues(IList<string>? values, bool strict, [NotNu
                         {
                             if (results == null)
                             {
-                                results = new List<T>();    // Allocate it only when used 
+                                results = new List<T>(4);   // Allocate it only when used. 4 is the default capacity after 1 item is added
                             }
```

Comment:
```c#
results ??= new List<T>(4); // ... comment
```
?

Reply:
Happy to simplify the syntax if asked for. I didn't want to change too much that wasn't directly related to the PR intention.